lib/array.rb

```diff
@@ -0,0 +1,52 @@
+class Array
+
+
+  def my_inject(sum=first,method=nil)
+
+    if method == nil && sum != first
+      if !block_given?
+      method = sum
+      sum = first
+      end
+    end
+    if method == nil
+      raise 'no block given' if !block_given? && method == nil
+      each do |x|
+        if index(x) == 0 && sum != first
+          sum = yield sum,x
+        end
+        if index(x) != 0
+          sum = yield sum,x
```

Comment:
If you see repetition, as in lines 17 and 19, and again lines 26 and 30, think about how you can condense this into something more readable. Maybe line 15 could read `sum = yield sum, x if (index(x) == 0 && sum != first) || index(x) != 0` - still quite verbose, but maybe easier to reason about